Dockerfile

```diff
@@ -8,16 +8,21 @@ RUN apk update && \
 
 # Copy our source code into the container.
 WORKDIR /go/src/github.com/chihaya/chihaya
-ADD . /go/src/github.com/chihaya/chihaya
+COPY . /go/src/github.com/chihaya/chihaya
 
 # Install our golang dependencies and compile our binary.
 RUN glide install
-RUN go install github.com/chihaya/chihaya/cmd/chihaya
+RUN CGO_ENABLED=0 go install github.com/chihaya/chihaya/cmd/chihaya
+RUN adduser -D chihaya
 
-# Delete the compiler from the container.
-# This makes the container much smaller when using Quay's squashing feature.
-RUN rm -r /usr/local/go
+FROM scratch
```

Comment:
I've been burned by using scratch with Go binaries in the past, if you aren't extremely careful about how you compile. For example, by default Go links to libc and uses it for networking (iirc DNS resolution), so putting a binary in scratch without compiling with the `netgo` tag passed to the linker can cause subtle problems.

We can just use `alpine:latest` for the runtime container to avoid some of this complexity and so that we're running the same environment in which it was built, just without the filesystem overhead of the development toolchain.

Reply:
Scratch image works pretty well for me, but I don't mind to do alpine based Dockerfile in a day or two.

Reply:
Yeah, I suppose Chihaya doesn't use any DNS resolution, but there's a chance in the future some middleware could use it.

Reply:
Forgot to thank you for sharing this, didn’t know about that caveat of scratch images :) So if I just change “FROM scratch” to “FROM alpine” we will be fine?